functions/APIs/deals.js

```diff
@@ -0,0 +1,115 @@
+const { db } = require('../util/admin');
+
+exports.getAllDeals = (request, response) => {
+    db.collection('deals').get()
+    .then((data) => {
```

Comment:
Are you using a transpiler like Babel or Webpack for this code?  I know the React code is using Babel, but is this as well?  If so, you can use the `async` and `await` keywords, which make this flow a lot more naturally.

Reply:
I will try using the `async` and `await` keywords to query multiple databases.

Reply:
Given the time pressure, let's hold off on this until a future PR.